enhancements/cluster-logging/cluster-logging-v2-apis.md

```diff
@@ -0,0 +1,429 @@
+* Drop support in the **cluster-logging-operator** for **log-view-plugin** management 
+* Support log forwarder API with minimal dependency upon reserved words (e.g. default)
+* Support an API to spec a Red Hat managed LokiStack with the logging tenancy model
+* Support an API to allow flexible deployment of the logging components: collector/forwarder, storage, visualization
+* Continue to allow deployment of a log forwarder to the output sinks of the administrators choosing
+
+
+### Non-Goals
+
+* "One click" deployment of a full logging stack as provided by **ClusterLogging** v1
+* Complete backwards compatibility to **ClusterLogForwarder** v1
+* Automated migration path from v1 to v2
+
+## Proposal
+
+This is where we get down to the nitty gritty of what the proposal
+actually is. Describe clearly what will be changed, including all of
+the components that need to be modified and how they will be
+different. Include the reason for each choice in the design and
+implementation that is proposed here, and expand on reasons for not
+choosing alternatives in the Alternatives section at the end of the
+document.
+
+### Workflow Description
+
+The following workflow describes the first user story which is a superset of the others and allows deployment of a full logging stack to collect and forward logs to a Red Hat managed log store.
+
+**cluster administrator** is a human responsible for:
+
+* Managing and deploying day 2 operators
+* Managing and deploying an on-cluster LokiStack
+* Managing and deploying a cluster-wide log forwarder
+
+**obervability-operator** is an operator responsible for:
+
+* managing and deploying observability plugins (e.g log-view-plugin)
+
+**loki-operator** is an operator responsible for managing a loki stack
+
+**cluster-logging-operator** is an operator responsible for managing log collection and forwarding
+
+The cluster administrator does the following:
+
+1. Deploys the Red Hat **observability-operator**
+1. Deploys the Red Hat **loki-operator**
+1. Deploys an instance of **LokiStack** in the `openshift-logging` namespace
+1. Deploys the Red Hat **cluster-logging-operator**
+1. Creates a **ClusterLogForwarder** custom resource for the **LokiStack**
+
+The **loki-operator**:
+
+1. Deploys the logging-view-plugin for reading logs in the OpenShift console
+
+The **cluster-logging-operator**:
+
+1. Deploys the log collector to forward logs to log storage in the `openshift-logging` namespace
+
+### API Extensions
+
+CustomResourceDefinition:
+```yaml
+    apiVersion: "logging.openshift.io/v2"
+    kind: ClusterLogForwarder
+    metadata:
+      name: 
+    spec:
+      serviceAccount:
+        name: 
+        namespace:       #namespace of deployment and resources
+      inputs:
+      - name:
+        type:
+        application:
+          selector:  #labelselector
+          namespaces:
+            include: []
+            exclude: []
+          containers:
```

Comment:
Why we do not use pods?
          pods:
            include: []
            exclude: []

Reply:
@anpingli in addition to containers or instead of containers?  Pods are better selected IMO by labels which is the construct we should promote.  There is no mechanism for choosing specific containers in a kubernetes way

Reply:
"instead of containers",  depend on this feature https://issues.redhat.com/browse/LOG-1296

Reply:
The construct mimics what we are delivering as part of 5.8 to allow you to selectively include/exclude containers.  Some customers have expressed the desire to not collect logs from containers.  As stated in the previous comment, pods are a concept when querying the API server and containers are not.  There is no way to choose a container based upon a label.  Once LOG-2155 lands, I would propose there is no longer a need for the annotation of LOG-1296 given you can now define separate inputs to accomplish the same result.  Testing and usage will determine if this assumption is accurate.

Reply:
@anpingli Pods typically have randomized names that aren't very useful as an identifier. Container names are (usually) stable names for the application running in the container, which is useful for telling what kind of logs will be produced.